data/org.freedesktop.impl.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,156 @@
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>names a{sv}</term>
+            <listitem><para>
+              The different shortcut names that have been registered mapped to user-readable descriptions for the app to present that describes the shortcut.
+            </para></listitem>
+          </varlistentry>
+      </variablelist>
+    -->
+    <method name="ListShortcuts">
+      <arg type="o" name="handle" direction="in"/>
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="a{sv}" name="results" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out1" value="QVariantMap"/>
+    </method>
+
+    <!--
+        Bind:
+        @handle: Object path for the #org.freedesktop.impl.portal.Request object representing this call
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @shortcut_id: The identifier of the shortcut we intend to register
+        @results: Vardict with the results of the call
+
+        Allows to configure the binding the global shortcut named @p shortcut_id within the @p session_handle session to a new shortcut.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>description</term>
+            <listitem><para>
+              The user-readable description of the registered binding. This is intended for the app to be able to render the shortcut into the UI, allowing the user to learn the shortcut
+            </para></listitem>
+          </varlistentry>
+      </variablelist>
+    -->
+    <method name="Bind">
+      <arg type="o" name="handle" direction="in"/>
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="shortcut_id" direction="in"/>
+      <arg type="a{sv}" name="results" direction="out"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.Out1" value="QVariantMap"/>
+    </method>
+
+    <!--
+        Activated:
+        Notifies about a shortcut becoming active.
+
+        @session_handle: Session that requested the shortcut
+        @shortcut_id: the application-provided ID for the notification
+    -->
+    <signal name="Activated">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="shortcut_id" direction="in"/>
+    </signal>
+
+    <!--
+        Deactivated:
+        Notifies that a shortcut is not active anymore.
+
+        @session_handle: Session that requested the shortcut
+        @shortcut_id: the application-provided ID for the notification
+    -->
+    <signal name="Deactivated">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="shortcut_id" direction="in"/>
+    </signal>
```

Comment:
activated/deactivated indicates that applications care about the state, equivalent to press/release. Which also suggests that applications *may* care about the duration of that state for things like key repeat. Is that the intention here? If not, a single signal `Triggered` may be sufficient?

Reply:
It's important to support use-cases like Push-to-Talk